scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -94,6 +93,11 @@ def __init__(self, instance, type_str="build"):
         self.args = []
         self.terminated = False
 
+    def test_timeout(self):
+        return math.ceil(self.instance.testsuite.timeout *
+                         self.instance.platform.timeout_multiplier *
```

Comment:
Isnt this going to "exponentially" increase the timeout multiplying this again by whatever the platform has?

So if I have timeout of 100 and platform has a 4 multiplier and I use the new argument with 4, it will be 100 * 4 * 4.

I think we need to ignore what platform has provided if multiplier is provided on command line.

Reply:
Not sure, then running a simulation of different platforms on a 'slow' host would have the same platform penalty for all platforms (e.g. single and multi-core ISS). They feel like different things to me, better apply them both.

Reply:
> This is especially useful in case of simulation platform as simulation time may depend on the host speed & load, we may select different simulation method (i.e. cycle accurate but slower one), etc...

`self.instance.platform.timeout_multiplier` was added to address this issue on a platform level, now we are adding an option that is used on the global level, they are mutually exclusive IMO.

Reply:
For 'real' platforms indeed that works. The issue addressed here, however, is different, mostly for simulation platforms. It is not about the relative speed/performance of a platform (e.g. fast or slow simulator), but about the speed of the machine running the simulator. Current platforms have been calibrated for Zephyr CI, but on other environments may need 'tuning'. That's about the development host, not about the specific platform being simulated.

Reply:
ok, my point here is that you use this option only with a set of platforms, not when you call twister globally, i.e. 

`twister -p nsim_hs5x_smp_12cores --timeout-multiplier 10`

but looking in the tree:

```
identifier: nsim_hs5x_smp_12cores
name: Multi-core HS5x nSIM simulator
type: sim
simulation: mdb-nsim
simulation_exec: mdb
arch: arc
toolchain:
  - zephyr
  - arcmwdt
  - cross-compile
supported:
  - smp
testing:
  timeout_multiplier: 4
  ignore_tags:
    - net
    - bluetooth
```

There is already a multiplier, so the timeout will be multiplied again by 4 (effective multiplier will be 40). Why not call twister with the value of 40 instead and avoid having to do special calculations?

Anyways, it is time you will spend waiting for something to timeout unnecessarily IMO.

Reply:
right, this example is about a 12-core SMP simulation target. We run that together with a 1-core ARC simulation in our internal regression testing environment, or maybe on a local developer PC. the x4 for the platform handles the 12 vs 1 core simulation speed. The global parameter handles the difference between running nSIM on local PC, Synopsys compute farm, or Zephyr CI.

Reply:
@nashif 

> I think we need to ignore what platform has provided if multiplier is provided on command line.

It's intentionally implemented this way otherwise it would be really counterintuitive for user. I.e. we may want to increase timeout, but adding `--timeout-multiplier` may decrease timeout  (if the value in board.yaml is bigger than the value we've selected via `--timeout-multiplier`). It's even worse if several platforms are selected as we may increase timeouts for one and decrease timeouts for another.

So yes, we should use it as a coeficient rather than overrde platform provided value.

Reply:
> It's even worse if several platforms are selected as we may increase timeouts for one and decrease timeouts for another.

Do not understand how is the total timeout is  going to decrease, can you explain?

Reply:
then, in this case, do:

`twister -p board_slow -p board_fast --timeout-multiplier 2`, i.e. use the the multiplier of the slowest.

Reply:
> PS: I don't see any reason to override `timeout_multiplier` value in board configuration with twister option. They serve diferent purpose - one for tweaking timeout because of platform specifics (i.e. slow HW / simulator) another one for tweaking timeout because of host / launch specifics (significant host load due to overcommit, extra delay for simulator / debugger licence check, different simulation method usage / etc...)

The timeout will mostly be impacted by the simulator and less by the host being slow. If the host is slow, it does not mean  all tests will run slower.